python/cuml/cuml/ensemble/randomforestclassifier.pyx

```diff
@@ -828,3 +855,35 @@ class RandomForestClassifier(BaseRandomForestModel,
         if self.dtype == np.float64:
             return get_rf_json(rf_forest64).decode('utf-8')
         return get_rf_json(rf_forest).decode('utf-8')
+
+    def cpu_to_gpu(self):
+        # treelite does an internal isinstance check to detect an sklearn
+        # RF, which proxymodule interferes with. We work around that
+        # temporarily here just for treelite internal check and
+        # restore the __class__ at the end of the method.
+        if GlobalSettings().accelerator_active:
+            cls_cahed = self._cpu_model.__class__
+            self._cpu_model.__class__ = sys.modules['sklearn.ensemble'].RandomForestClassifier
+
+        super().cpu_to_gpu()
+
+        if GlobalSettings().accelerator_active:
+            self._cpu_model.__class__ = cls_cahed
+
+    @classmethod
+    def _hyperparam_translator(cls, **kwargs):
+        kwargs, gpuaccel = super(RandomForestClassifier, cls)._hyperparam_translator(**kwargs)
+
+        if "max_samples" in kwargs:
+            if isinstance(kwargs["max_samples"], int):
+                logger.warn(
+                    f"Integer value of max_samples={kwargs['max_samples']}"
+                    "not supported, changed to 1.0."
+                )
+                kwargs["max_samples"] = 1.0
```

Comment:
If the integer is 0, is this still correct? Not even sure if an integer value of 0 makes sense.

Reply:
It is not allowed, even the fp range is `(0.0, 1.0]`